### ISSUE TITLE
[Need a fix/suggestion] Altering a timestamp column to timestampz column doesn't reflect in postgres

### ISSUE THREAD

**Issue by dashsatish** [piccolo-orm/piccolo#56]
Hi @dantownsend 
I have a `Date` column/field which showed postgres datatype as `timestamp`. After altering the field type in the model to use
`Timestamptz`, it still shows `timestamp` instead of `timestamptz`. Can you please check. It also breaks insert statements.

**Comment by dantownsend** [comment]
@dashsatish That's strange - I'll investigate and try and find a fix.

**Comment by dashsatish** [comment]
I even see for floats the postgres type is `numeric` when created newly but on a case where there is a model change and we're changing the type of the field to `float`. It comes out as `float8` in the db which is a bit unusual.

On wider note could you please revisit on the column type classes once:
- Timestamp
- Timestamptz
- Float

(Irrespective of anything, I appreciate the amount and efforts you have put in :). Thanks a ton for that)

**Comment by dantownsend** [comment]
@dashsatish I've been able to replicate this - working on a fix. Seems to be a general issue, where Piccolo can't convert column types in migrations.

**Comment by dantownsend** [comment]
@dashsatish This should be fixed now in the latest version on PyPI - 0.17.0.

Migrations now detect when the column type has changed, and will try and update it in Postgres.

Postgres can't automatically convert all column types - for example, an `Integer` to a `Timestamp`, but it should work for most common use cases (e.g. `Varchar` to `Text`).

The fix works for new migrations - existing migrations don't contain enough information. They just need two extra attributes - `column_type` and `old_column_type`. Here's an example:

```python
from piccolo.apps.migrations.auto import MigrationManager
from piccolo.columns.column_types import Timestamp
from piccolo.columns.column_types import Timestamptz

ID = "2021-02-14T15:25:36"
VERSION = "0.17.0"

async def forwards():
    manager = MigrationManager(migration_id=ID, app_name="home")

    manager.alter_column(
        table_class_name="Task",
        tablename="task",
        column_name="created_on",
        params={},
        old_params={},
        column_class=Timestamptz,  # New attribute
        old_column_class=Timestamp,   # New attribute
    )
```
The best solution for old migrations is to add these attributes by hand.

Thanks for reporting this issue, and let me know if you still experience problems with it.

**Comment by dashsatish** [comment]
This can be closed we are good here. If there are any other problems I would raise a hand. Thanks a lot.

**Comment by dashsatish** [comment]
Fixed as of 0.72.0!
